apps/redi-connect/src/pages/app/find-a-mentor/FindAMentor.tsx

```diff
@@ -161,6 +163,17 @@ const FindAMentor = () => {
     })
   )
 
+  const menteeHasAnActiveMatch =
+    myProfileQuery.data?.conProfile?.userType === 'MENTEE' &&
+    myProfileQuery.data?.conProfile?.mentorshipMatches.length > 0 &&
+    myProfileQuery.data?.conProfile?.mentorshipMatches?.[0].status ===
+      'ACCEPTED'
+
+  if (menteeHasAnActiveMatch) {
+    const matchId = myProfileQuery.data?.conProfile?.mentorshipMatches?.[0].id
+    history.replace(`/app/mentorships/${matchId}`)
+  }
+
```

Comment:
We seem to be accessing information inside conProfile a lot of times with this change (and even including below in the if check before redirecting to LoggedIn.

How about reordering and rewriting this a bit? My recommendation would be something like the following: (Example code, might not work but just to give the idea)

```js
const profile = myProfileQuery.data?.conProfile;

// First let's check if this profile is even approved
if (conProfile && conProfile.profileStatus !== ConnectProfileStatus.Approved)
  return <LoggedIn />

// Then the active mentorship check
const menteeHasAnActiveMatch = conProfile?.userType === 'MENTEE' && conProfile?.mentorshipMatches.length > 0 && conProfile?.mentorshipMatches?.[0].status === 'ACCEPTED'

if (menteeHasAnActiveMatch) {
  // Do the history replace
}
```

What do you think @katamatata?

Reply:
Totally agree. Thank you!